dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -103,64 +105,64 @@ def __add__(self, other):
     def update_profile(self, df_series, pool=None):
         """
         Updates the profiles from the data frames
-        
+
         :param df_series: a given column, assume df_series in str
         :type df_series: pandas.core.series.Series
         :param pool: pool to utilized for multiprocessing
         :type pool: multiprocessing.Pool
         :return: Self
-        :rtype: BaseColumnProfileCompiler
+        :rtype: BaseCompiler
         """
-        
-        if len(self._profilers) == 0:
-            return 
-        
+
+        if not self._profilers:
+            return
+
         # If single process, loop and return
         if pool is None:
-            for col_profile in self._profiles:
-                self._profiles[col_profile].update(df_series)
+            for profile_key in self._profiles:
+                self._profiles[profile_key].update(df_series)
             return self
-        
+
         # If multiprocess, setup pool, etc
         single_process_list = []
         multi_process_dict = {}
-        
-        # Spin off seperate processes, where possible
-        for col_profile in self._profiles:
-            
-            if self._profiles[col_profile].thread_safe:
-                
-                try: # Add update function to be applied on the pool
-                    multi_process_dict[col_profile] = pool.apply_async(
-                        self._profiles[col_profile].update, (df_series,))
-                except Exception as e: # Attempt again as a single process
-                    self._profiles[col_profile].thread_safe = False
-                
-            if not self._profiles[col_profile].thread_safe:                
-                single_process_list.append(col_profile)
+
+        # Spin off separate processes, where possible
+        for profile_key in self._profiles:
+
+            if self._profiles[profile_key].thread_safe:
+
+                try:  # Add update function to be applied on the pool
+                    multi_process_dict[profile_key] = pool.apply_async(
+                        self._profiles[profile_key].update, (df_series,))
+                except Exception as e:  # Attempt again as a single process
+                    self._profiles[profile_key].thread_safe = False
+
+            if not self._profiles[profile_key].thread_safe:
+                single_process_list.append(profile_key)
```

Comment:
I think `profile_key` even further abstracts the confusing nature of what a "profile" is, IMO. 

I think @JGSweets mentioned something such as `inspectors` (so, it'd be `int_inspector` or `categorical_inspector`). 

Also instead of `profile_key` maybe `profile_type` or something makes more sense.

Reply:
I personally think `profile_type` is considerably more clear

Reply:
done